### PR TITLE
feat: add responsive icons

### PR DESCRIPTION
**Pull request by mkrl**
[FX-4267](https://toptal-core.atlassian.net/browse/FX-4267)

**best reviewed on per-commit basis, one commit only includes generated files**

### Description

This PR adds responsive variants for all the icons + the related generator.
Full guidelines for standalone responsive buttons will be added in the follow-up PR.

### How to test

- [Temploy](https://picasso.toptal.net/fx-4179-create-responsive-icons)
- `Icon` docs in Picasso should have a new story called `responsive`
- Generator script looks good
- Generator script adds new responsive icons you are able to commit when adding new icons

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [n/a] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4267]: https://toptal-core.atlassian.net/browse/FX-4267?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mkrl** [comment]
@toptal-anvil ping reviewers

**Comment by augustobmoura** [comment]
Do we have some statistics on how this would affect final bundles, if affect at all? For example building staff-portal with it

**Comment by mkrl** [comment]
@augustobmoura it should not affect the size unless they are used (exactly the same way as normal icons).

File: packages/picasso/src/Icon/index.ts [hunk 1]
```diff
@@ -404,3 +404,205 @@ export { default as WhitePaper16 } from './WhitePaper16'
 export { default as WhitePaper24 } from './WhitePaper24'
 export { default as WorkExperience16 } from './WorkExperience16'
 export { default as WorkExperience24 } from './WorkExperience24'
+
+export { default as BoldResponsive } from './BoldResponsive'
```
Comment (dmaklygin) [inline packages/picasso/src/Icon/index.ts:408]: didn't we consider to call them without postfix `Responsive` ?

**Review by dmaklygin — COMMENTED** [review]
I thought we decided to not add a suffix `Responsive` to responsive Icons.
I think, it would be better to know highlight a feature that an icon is responsive

**Comment by augustobmoura** [comment]
> @augustobmoura it should not affect the size unless they are used (exactly the same way as normal icons).

Maybe we could have a draft PR on SP and CP to make sure that doesn't happen, I know that it should behave normally, but if it doesn't it would cause a significant increase in bundle size on the projects

File: .changeset/tidy-cheetahs-call.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+'@toptal/picasso': minor
+---
+
+### Icon
+
+- added responsive icons (24px on xl screens, 16px otherwise)
```
Comment (sashuk) [inline .changeset/tidy-cheetahs-call.md:7]: Is it vice-versa?

Comment (mkrl) [reply .changeset/tidy-cheetahs-call.md]: updated, thank you

**Comment by mkrl** [comment]
@toptal-bot run package:alpha-release

**Review by sashuk — APPROVED** [review]
Code looks good (checked couple of generated components as well)


https://github.com/toptal/picasso/assets/1390758/3732850e-04b7-41f3-962d-a10a2a5c3481

File: bin/generate-responsive-icons.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    )
+  })
+}
+
+const insertExports = (iconNames, indexFilePath) => {
+  const exportLines = iconNames.map(
+    item => `export { default as ${item}Responsive } from './${item}Responsive'`
+  )
+
+  fs.readFile(indexFilePath, 'utf-8', (err, data) => {
```
[the remark this answers is not in the document]
Comment (mkrl) [reply bin/generate-responsive-icons.js]: updated all fs operations to async

**Review by augustobmoura — APPROVED** [review]
LGTM! Nice work :+1:

**Pull request merged**